lib/triangle.h

```diff
@@ -31,12 +31,13 @@ class Triangle {
         , reflectivity(reflectivity)    // reflectivity factor
         , u(vertices[1] - vertices[0])
         , v(vertices[2] - vertices[0])
-        , normal(u^v)
         , uv(u*v)
         , vv(v*v)
         , uu(u*u)
         , denom(uv * uv - uu * vv)
-    {}
+    {
+        normal = (u^v).Normalize();
```

Comment:
You can do it directly in the initializer, since .Normalize() also returns the vector.